### PR TITLE
Code refactored

### PR DESCRIPTION
**Pull request by Domejko**
## Description

I have done code refactoring #160. As I mentioned in comment I did shorten some nested `if` statements and when it was possible and clean looking made them one liners. Added a specific expected errors to the `except` blocks. In **auth_backends.py** I did change code accordingly because either way if `user.check_password(password)` was successful it did return `user`:

Before: 
```py
if user.check_password(password):
    if user.is_active:
        return user

    if user.awaiting_email_verification:
        return user
    return user
```
After:
```py
if user.check_password(password):
    return user
```
From file **create.py** in */backend/views/core/invoices* I did delete function named `edit_invoice_page` because it was exactly the same as this function:
```py
@require_http_methods(["GET", "POST"])
def create_invoice_page(request: HttpRequest):
    if request.method == "POST":
        return invoice_page_post(request)
    return invoice_page_get(request)
```
and it have not been used anywhere in the code, so this function can be renamed to let say `create_edit_invoice_page` if would be needed for editing purposes anywhere in future.

I did noticed that some functions like `edit_invoice` are repeating through out the code with some slight changes here and there which possibly could be reduced to one function and repurposed with additional arguments. If you like I could try to work on that.

Hope you will be satisfied with those small changes.

# Checklist

- [x] Ran the [Black Formatter](https://docs.strelix.org/MyFinances/#/how-to-contribute?id=test-and-lint) and
  [djLint-er](https://docs.strelix.org/MyFinances/#/how-to-contribute?id=test-and-lint) on any new code
  (checks
  will
  fail without)
- [x] Made any changes or additions to the documentation _where required_
- [x] Changes generate no new warnings/errors
- [x] New and existing [unit tests](https://docs.strelix.org/MyFinances/#/how-to-contribute?id=test-and-lint) pass locally with my
  changes

## What type of PR is this?

- ♻️ Code Refactor

## Added/updated tests?

- 🙅 no, because they aren't needed

## Related PRs, Issues etc
- Related Issue #
- Closes #

### CODE REVIEW

File: backend/views/core/invoices/create.py [hunk 1]
```diff
@@ -55,10 +55,7 @@ def invoice_page_post(request: HttpRequest):
         invoice.client_city = request.POST.get("to_city")
         invoice.client_county = request.POST.get("to_county")
         invoice.client_country = request.POST.get("to_country")
-        if request.POST.get("is_representative") == "on":
-            invoice.client_is_representative = True
-        else:
-            invoice.client_is_representative = False
+        invoice.client_is_representative = True if request.POST.get("is_representative") == "on" else Fals
```
Comment (TreyWW) [inline backend/views/core/invoices/create.py:58]: Can this bug be fixed please - `False` instead of `Fals`

Comment (Domejko) [reply backend/views/core/invoices/create.py:58]: Yes, of course, I just did noticed that and will fix it right away

Comment (Domejko) [reply backend/views/core/invoices/create.py]: Typo fixed and now all checks passed

File: backend/auth_backends.py [hunk 2]
```diff
@@ -18,10 +18,5 @@ def authenticate(self, request, username=None, password=None, **kwargs):
             return None
         else:
             if user.check_password(password):
-                if user.is_active:
-                    return user
-
-                if user.awaiting_email_verification:
```
Comment (TreyWW) [inline backend/auth_backends.py:24]: This was going to initially be used to actually send the user to a different page if still awaiting email verification (or at least send a notification), but that's my fault for not adding a comment or anything. I'll just write myself a note to add this later

**Review by TreyWW — APPROVED** [review]
Great cleanup! Really appreciate your contribution :)

**Pull request merged**